### PR TITLE
[Feature] App Settings

### PR DESCRIPTION
**Pull request by Mikunj**
This PR adds 2 new app specific settings:
- Starting page selection
- Data saver mode

### Starting page selection
Starting page selection allows the user to choose between `Feed`, `Search`, `Quick Update` and `Library` as their starting page.

### Data saver
Data saver mode makes it so any image embeds in the feed requires a tap before it is shown to the user. This has 2 advantages:
- Limits data usage for those with low mobile data
- Drastically improves performance for lower-end devices as images aren't being rendered.

We could enable this by default on lower end devices if we want.

screen shot 2018-06-07 at 2 26 26 pm

### Drawbacks
Both these settings are stored using redux persist, thus if persist fails then these settings revert to their default values and users will have to re-set their settings.

The hack job in the navigation may also have some unforeseen drawbacks but from my quick testing session everything seemed to work as intended.

### TODO
- [x] Improve the look of tap to load image
- [x] Test on android
- [x] Test on tablets

### CODE REVIEW

File: src/screens/Sidebar/AppSettings.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    return (
+      <View style={styles.containerStyle}>
+        <ScrollView scrollEnabled={false}>
+          <SidebarTitle title={'General'} />
+          <SelectMenu
+            style={styles.selectMenu}
+            onOptionSelected={this.onStartingPageChange}
+            cancelButtonIndex={pages.length - 1}
```
Comment (trmcnvn) [inline src/screens/Sidebar/AppSettings.js]: I don't think this needs to be defined as the default is `-1` which will be the last element in the list.

File: src/routes/tabs.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import React, { PureComponent } from 'react';
+/* eslint react/no-multi-comp:0 */
```
Comment (trmcnvn) [inline src/routes/tabs.js]: I believe these changes will need to be made to our `tabNotification.js` file as well.

Comment (Mikunj) [reply src/routes/tabs.js]: do we need to? the top of the tab notification says the following:
```
Created a hacky TabNavigator with initial screen of Notifications. This way user can navigate to related notification.
```

Comment (trmcnvn) [reply src/routes/tabs.js]: I meant more the `DrawerWrapper` content but if that isn't applicable. that's fine :+1:

**Pull request merged**